vault/data_source_aws_static_credentials_test.go

```diff
@@ -0,0 +1,55 @@
+package vault
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/testutil"
+	"testing"
+)
+
+func TestAccDataSourceAWSStaticCredentials(t *testing.T) {
+	_, _ = testutil.GetTestAWSCreds(t)
```

Comment:
Related to this, which looks a little weird, I imagine:

I found if i included `access_key` and `secret_key` in the backend specification for the test, it would break vault's ability to get information from IAM ("Invalid security token"). I'm not sure whether this is a bug with my local vault, a possible issue in the static roles implementation in vault, or some other weird local artifact.

If I run solely with access credentials set via env var, things work.

Reply:
Ah yes, I think this has something to do with how the AWS Secrets Engine always needs the STS token passed as an environment variable. 

Going to cc @calvn since they recently confirmed this behavior in the AWS Secrets Engine when the team was working on the static roles implementation for Vault. It doesn't look like token info can be passed to the AWS Config Root endpoint, is that correct?

Reply:
I looked back at the investigation and this is what I found: There is not a way to provide the session token to the AWS secrets engine to the config other than the aws env var. This is likely intentional since sts creds are not supposed to be long lived and therefore not intended to be used as creds you provide to the root config endpoint (outside of testing purposes). The typical recommendation is to provide IAM user creds and then rotate root creds.

That being said @kpcraig can you check to see if you have any env variables set in your shell session where you vault server and/or TF tests are running? Maybe these are causing issues?

Reply:
As a followup, it seems like the minimum version that works is having env-vars set in the vault environment, although setting actively incorrect settings in a terraform file will break things.

Reply:
Should we add a comment explaining why we are calling GetTestAWSCreds but not using the return values?

Reply:
Also, I don't think I quite follow what the problem is. I see in resource_aws_secret_backend_role_test.go we are passing the values to the config: https://github.com/hashicorp/terraform-provider-vault/blob/716f90065eda7d13a11b87decf0ae53f259556b5/vault/resource_aws_secret_backend_role_test.go#L35. What is the difference here?

Reply:
I have found that in this case anyway, setting the backend values at all breaks the test, even if they are correct, and match those that the vault instance knows.

Reply:
So, after some investigation, it looks like in some cases the Access/Secret pair is insufficient, and the method I was using to obtain credentials required a session token for access. Using my personal AWS account, I was able to get it working, so I've re-added the access/secret fields for the aws backend in the tests.